live/live-root/usr/lib/dracut/modules.d/99live-self-update/live-self-update-parser.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# redirect stdout/stderr to systemd journal using the "live-self-update"
+# identifier, later the self-update logs can be filtered by using command
+# "journalctl -t live-self-update" which includes the output from this dracut
+# hook and the self-update service as well
+exec > >(systemd-cat -t live-self-update) 2>&1
+
+. /lib/dracut-lib.sh
+
+FILE="/etc/cmdline.d/agama-self-update-generated.conf"
+REPO_FILE="/run/live-self-update/repositories"
+
+mkdir -p $(dirname "$REPO_FILE")
+
+get_url() {
+  # Agama driver updates
+  local url
+  url=$(getargs "inst.self_update=")
+
+  if [ "$url" == "0" ]; then
+    echo "<5>Self update disabled via boot parameter"
+  elif [ -z "$url" ]; then
```

Comment:
so far it **LGTM** as it does not setup network by default an only if **inst.sel_update** is provided. Once we enabled it by default we should consider cases were the network is disabled explicitly (**rd.neednet=0**) as well as the default **NetworkManager** profile generated by **nm-initrd-generator** which is a multi-connect profile.

Last but not least, maybe we should handle also the proxy option passed as a kernel boot option although it can be part of the repository URL but just for avoiding having to declare it twice.